### ISSUE TITLE
Allow not uploading any file

### ISSUE THREAD

**Issue by joelpet** [jrbasso/MeioUpload#6]
Is it possible to let the filename field be left empty on submission and in that case not upload anything?

I've been trying to achieve that for a while now, without any success. I managed to disable 'Empty' validation. However, it gives an a notice about 'Uninitialized string offset: 0' for meio_upload.php, line 634.

The reason for wanting this is simply that I have other fields, and the file field is thereby not required by me.

**Comment by jrbasso** [comment]
Can you write a test or steps to reproduce?

**Comment by joelpet** [comment]
I've created a couple of files (not sure how to set up tests and fixtures, yet) necessary to reproduce, including:

Model: http://empty-meioupload.pastebin.com/f3ccbc829
Controller: http://empty-meioupload.pastebin.com/f7968adb5
Add View: http://empty-meioupload.pastebin.com/f4b93e0b8
Index View: http://empty-meioupload.pastebin.com/fd4e9d67
SQL setup: http://empty-meioupload.pastebin.com/f44283270

Make sure debug level is set to 1 at minimum. Now, try to add an EmptyTest without selecting any file for 'Filename'. If I do that, I end up with the following notices, http://empty-meioupload.pastebin.com/f336b63fa

The EmptyTest is indeed saved, however 'Dir' is filled in and 'Filename' is set to '0.', '1.', '2.' and so on...

**Comment by joelpet** [comment]
I came up with the following quick fix. I can't guarantee that it doesn't break anything, even though it feels like it shouldn't. Anyway, it seems to do the trick for me and I thought maybe it would inspire you. :) 

http://empty-meioupload.pastebin.com/f5814305e

Besides, it might also solve issue #9. At least, after applying the patch above I didn't experience the problems described in issue #9.

**Comment by jrbasso** [comment]
Thanks joelpet. I pushed the code with your changes. Can you test to see if all is ok? Thanks.

**Comment by joelpet** [comment]
Glad to help! It seems to be functioning as before, with the only difference that it doesn't spit out notices when submitting a MeioUpload form without selecting any file. However, I am still experiencing some trouble concerning the thumbsizes addressed in issue #8, but that doesn't seem related to this.

**Comment by joelpet** [comment]
So, I looked into this a bit more because I needed a way to delete the uploaded file and adherent meta-data in database while keeping the record as for the rest. I came up with the following changes.

http://gist.github.com/177243
http://gist.github.com/177246

These do also include a fix for the thumbnails directories, where the `normal` thumb directory was erroneously created and files in the thumb directories were not deleted. However, here you might want to change so that only specified thumbsize directories are searched in, which should be faster than a recursive search.
